daemons/based/based_messages.c

```diff
@@ -157,40 +157,35 @@ cib_process_ping(const char *op, int options, const char *section, xmlNode * req
     const char *seq = crm_element_value(req, F_CIB_PING_ID);
     char *digest = calculate_xml_versioned_digest(the_cib, FALSE, TRUE, CRM_FEATURE_SET);
 
-    static struct qb_log_callsite *cs = NULL;
-
     crm_trace("Processing \"%s\" event %s from %s", op, seq, host);
     *answer = create_xml_node(NULL, XML_CRM_TAG_PING);
 
     crm_xml_add(*answer, XML_ATTR_CRM_VERSION, CRM_FEATURE_SET);
     crm_xml_add(*answer, XML_ATTR_DIGEST, digest);
     crm_xml_add(*answer, F_CIB_PING_ID, seq);
 
-    if (cs == NULL) {
-        cs = qb_log_callsite_get(__func__, __FILE__, __func__, LOG_TRACE,
-                                 __LINE__, crm_trace_nonlog);
```

Comment:
I just realized how crm_trace_nonlog fits into this. It's a "non-logging tracepoint" basically for the same situation as `pcmk__if_tracing()`, i.e. a way of doing something other than logging based on whether tracing is enabled. We should use it in the `pcmk__if_tracing()` definition instead of 0 (it looks like libqb treats 0 as "filter for logs", so skipping that will be a bit more efficient I guess).

Reply:
That makes sense, I see that now that you mention it. (I'm taking your word for the 0 behavior rather than going down the libqb rabbit hole.)

I guess `pcmk__apply_acl()`, where logging is part of the action, will still be fine since `crm_trace()` creates its **own** tracepoint.

Reply:
right